### PR TITLE
fixes uninstalls semantics to respect restarts and propagates autorev…

### PR DESCRIPTION
**Pull request by ProofOfKeags**
…ert changes to FE

### CODE REVIEW

File: agent/src/Lib/Algebra/Domain/AppMgr.hs [hunk 1]
```diff
@@ -65,9 +65,8 @@ data InfoRes a = InfoRes
           :: Include
               (Either_ (DefaultEqSym1 'OnlyDependencies) (ElemSym1 'IncludeDependencies) a)
               (HM.HashMap AppId DependencyInfo)
-    , infoResManifest
-          :: Include (Either_ (DefaultEqSym1 'OnlyManifest) (ElemSym1 'IncludeManifest) a) AppManifest
-    , infoResStatus :: Include (Either_ (DefaultEqSym1 'OnlyStatus) (ElemSym1 'IncludeStatus) a) AppContainerStatus
+    , infoResManifest :: Include (Either_ (DefaultEqSym1 'OnlyManifest) (ElemSym1 'IncludeManifest) a) AppManifest
+    , infoResStatus   :: Include (Either_ (DefaultEqSym1 'OnlyStatus) (ElemSym1 'IncludeStatus) a) AppContainerStatus
```
Comment (ProofOfKeags) [inline agent/src/Lib/Algebra/Domain/AppMgr.hs:69]: whitespace, ignore

**Review by dr-bonez — APPROVED** [review]
LGTM

File: agent/src/Handler/Apps.hs [hunk 2]
```diff
@@ -350,12 +353,17 @@ postUninstallAppLogic appId dryrun = do
         Just CreatingBackup  -> throwError (TemporarilyForbiddenE appId "uninstall" (show CreatingBackup))
         Just RestoringBackup -> throwError (TemporarilyForbiddenE appId "uninstall" (show RestoringBackup))
         _                    -> pure ()
-    let flags = if coerce dryrun then Left dryrun else Right (AppMgr2.Purge True)
-    breakageIds <- HM.keys . AppMgr2.unBreakageMap <$> AppMgr2.remove flags appId
-    bs <- pure (traverse (hydrate $ (AppMgr2.infoResTitle &&& AppMgr2.infoResVersion) <$> serverApps) breakageIds)
-        `orThrowM` InternalE "Reported app breakage for app that isn't installed, contact support"
+    let flags = if coerce dryrun then Left dryrun else Right (AppMgr2.Purge True, revert)
```
Comment (afg419) [inline agent/src/Handler/Apps.hs:356]: Need to support both dryrun + revert flags to give users a heads up before they sign off on revert

Comment (ProofOfKeags) [reply agent/src/Handler/Apps.hs]: I addressed this now.

**Comment by ProofOfKeags** [comment]
Theres some rust code in this PR. I'm not aware of how it got there, and I haven't dug into what it is doing. This still needs to be integration tested.

**Pull request merged**